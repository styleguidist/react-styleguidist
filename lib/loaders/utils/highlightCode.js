"use strict";

exports.__esModule = true;
exports.default = highlightCode;

var _glogg = _interopRequireDefault(require("glogg"));

var Prism = _interopRequireWildcard(require("prismjs"));

require("prismjs/components/prism-clike");

require("prismjs/components/prism-markup");

require("prismjs/components/prism-markdown");

require("prismjs/components/prism-css");

require("prismjs/components/prism-css-extras");

require("prismjs/components/prism-scss");

require("prismjs/components/prism-less");

require("prismjs/components/prism-javascript");

require("prismjs/components/prism-flow");

require("prismjs/components/prism-typescript");

require("prismjs/components/prism-jsx");

require("prismjs/components/prism-tsx");

require("prismjs/components/prism-graphql");

require("prismjs/components/prism-json");

require("prismjs/components/prism-bash");

require("prismjs/components/prism-diff");

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const logger = (0, _glogg.default)('rsg');
const IGNORED_LANGUAGES = ['extend', 'insertBefore', 'DFS'];

const getLanguages = () => Object.keys(Prism.languages).filter(x => !IGNORED_LANGUAGES.includes(x));
/**
 * Highlight code.
 *
 * @param {string} code
 * @param {string} lang
 * @returns {string}
 */


function highlightCode(code, lang) {
  if (!lang) {
    return code;
  }

  const grammar = Prism.languages[lang];

  if (!grammar) {
    logger.warn(`Syntax highlighting for “${lang}” isn’t supported. Supported languages are: ${getLanguages().join(', ')}.`);
    return code;
  }

  return Prism.highlight(code, grammar, lang);
}