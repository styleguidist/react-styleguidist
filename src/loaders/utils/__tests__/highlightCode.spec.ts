import glogg from 'glogg';
import highlightCode from '../highlightCode';

const logger = glogg('rsg');

const code = '<p>Hello React</p>';

it('should highlight code with specified language', () => {
	const actual = highlightCode(code, 'html');
	expect(actual).toMatchSnapshot();
});

it('should warn when language not found', () => {
	const warn = jest.fn();
	logger.once('warn', warn);

	const actual = highlightCode(code, 'pizza');
	expect(actual).toBe(code);
	expect(warn).toBeCalledWith(
		'Syntax highlighting for “pizza” isn’t supported. Supported languages are: plain, plaintext, text, txt, markup, html, mathml, svg, xml, ssml, atom, rss, css, clike, javascript, js, markdown, md, scss, less, flow, typescript, ts, jsx, tsx, graphql, json, webmanifest, bash, sh, shell, diff.'
	);
});

it('should not highlight code without language', () => {
	const actual = highlightCode(code);
	expect(actual).toBe(code);
});
