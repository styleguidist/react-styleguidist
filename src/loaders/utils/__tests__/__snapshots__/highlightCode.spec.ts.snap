// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should highlight code with specified language 1`] = `"<span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&lt;</span>p</span><span class=\\"token punctuation\\">></span></span>Hello React<span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&lt;/</span>p</span><span class=\\"token punctuation\\">></span></span>"`;
